File: memory-domain/src/main/java/com/memory/dto/search/MemorySearchResponse.java
package com.memory.dto.search;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;
import java.util.List;

@Getter
@Builder
public class MemorySearchResponse {

    private Long memoryId;
    private String title;
    private String content;
    private String locationName;
    private LocalDate memorableDate;
    private String memorableDateText;
    private String memoryType;
    private List<String> hashtags;
    
    // 메모리 작성자 정보
    private Long memberId;
    private String memberName;
    private String memberNickname;
    private String memberEmail;
    private String memberFileUrl;
    
    // 관계된 멤버 정보
    private Long relationshipMemberId;
    private String relationshipMemberName;
    private String relationshipMemberNickname;
    private String relationshipMemberEmail;
    private String relationshipMemberFileUrl;
    
    private HighlightInfo highlights;

    @Getter
    @Builder
    public static class HighlightInfo {
        private List<String> title;
        private List<String> content;
        private List<String> locationName;
        private List<String> hashtags;
        private List<String> memberName;
        private List<String> memberNickname;
        private List<String> relationshipMemberName;
        private List<String> relationshipMemberNickname;
    }
}

File: memory-domain/src/main/kotlin/com/memory/dto/search/MemorySearchResponse.kt
package com.memory.dto.search

import java.time.LocalDate

data class MemorySearchResponse(
    val memoryId: Long,
    val title: String,
    val content: String?,
    val locationName: String?,
    val memorableDate: LocalDate?,
    val memorableDateText: String?,
    val memoryType: String,
    val hashtags: List<String>?,

    // 메모리 작성자 정보
    val memberId: Long,
    val memberName: String,
    val memberNickname: String,
    val memberEmail: String,
    val memberFileUrl: String?,

    // 관계된 멤버 정보
    val relationshipMemberId: Long?,
    val relationshipMemberName: String?,
    val relationshipMemberNickname: String?,
    val relationshipMemberEmail: String?,
    val relationshipMemberFileUrl: String?,

    val highlights: HighlightInfo?
) {
    data class HighlightInfo(
        val title: List<String>?,
        val content: List<String>?,
        val locationName: List<String>?,
        val hashtags: List<String>?,
        val memberName: List<String>?,
        val memberNickname: List<String>?,
        val relationshipMemberName: List<String>?,
        val relationshipMemberNickname: List<String>?
    )
}
